tests/phpunit/CRM/Event/Form/ParticipantTest.php

```diff
@@ -55,8 +55,13 @@ public function testSubmitDualRole(): void {
         CRM_Core_PseudoConstant::getKey('CRM_Event_BAO_Participant', 'role_id', 'Volunteer'),
         CRM_Core_PseudoConstant::getKey('CRM_Event_BAO_Participant', 'role_id', 'Speaker'),
       ],
-    ])->postProcess()->getFirstMailBody();
-    $this->assertStringContainsString('Volunteer, Speaker', $email);
+    ])->postProcess();
+    $participant = \Civi\Api4\Participant::get()
+      ->addSelect('role_id:name')
+      ->addOrderBy('id', 'DESC')
+      ->execute()
+      ->first();
+    $this->assertEmpty(array_diff($participant['role_id:name'], ['Volunteer', 'Speaker']));
```

Comment:
I'm not sure array_diff is really what you want here because if only one role gets assigned (or no role), you want that to be a fail, but this will pass. Fortunately there is assertEqualsCanonicalizing which allows comparing arrays without order, so
```suggestion
    $this->assertEqualsCanonicalizing(['Volunteer', 'Speaker'], $participant['role_id:name']);
```

Reply:
@demeritcowboy thanks! updated